mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -5340,6 +5340,9 @@ class FoldContiguousGather final : public OpRewritePattern<GatherOp> {
   using OpRewritePattern::OpRewritePattern;
   LogicalResult matchAndRewrite(GatherOp op,
                                 PatternRewriter &rewriter) const override {
+    if (!op.getBase().getType().isa<MemRefType>())
+      return failure();
```

Comment:
[nit] Could you use `notifyMatchFailure`? Thanks!

Reply:
Good point, done.